src/ameedf/jb8223/assignments/RotationCheck.java

```diff
@@ -40,8 +40,93 @@
  *
  * Good luck !
  */
+
+import java.util.Scanner;
+
 public class RotationCheck {
-    public static void main(String[] args) {
-        // TODO: enter your code here :)
-    }
+	public static void main(String[] args) {
+		Scanner in = new Scanner(System.in);
+		int firstNumber = 0, secondNumber = 0;
+
+		// Get input from the user.
+		do {
+			System.out.print("Enter first number [0 to 1,000,000,000]:");
+			firstNumber = in.nextInt();
+		} while (firstNumber < 0 || firstNumber > 1000000000);
+
+		do {
+			System.out.print("Enter second number [0 to 1,000,000,000 and not equal to " + firstNumber + "]:");
+			secondNumber = in.nextInt();
+		} while (secondNumber < 0 || secondNumber > 1000000000 || firstNumber == secondNumber);
+
+		in.close();
+
+		// Calculate the length of the numbers
+		int firstNumberLength = 0, tempFirstNumber = firstNumber;
+		while (tempFirstNumber > 0) {
+			tempFirstNumber = tempFirstNumber / 10;
+			firstNumberLength++;
+		}
+
+		int secondNumberLength = 0, tempSecondNumber = secondNumber;
+		while (tempSecondNumber > 0) {
+			tempSecondNumber = tempSecondNumber / 10;
+			secondNumberLength++;
+		}
+
+		// Check the length of the numbers, if not equal then exit.
+		if (firstNumberLength != secondNumberLength) {
+			System.out.println("We cannot get from " + firstNumber + " to " + secondNumber + " by performing any number of rotations!");
+			return;
+		}
+
+		// Init arrays for numbers by required length
+		int[] firstNumberArray = new int[firstNumberLength];
+		int[] secondNumberArray = new int[firstNumberLength];
+
+		tempFirstNumber = firstNumber;
+		tempSecondNumber = secondNumber;
+
+		// Move from integers to arrays
+		for (int i = firstNumberLength; i > 0; i--) {
+			firstNumberArray[i - 1] = tempFirstNumber % 10;
+			tempFirstNumber = tempFirstNumber / 10;
+
+			secondNumberArray[i - 1] = tempSecondNumber % 10;
+			tempSecondNumber = tempSecondNumber / 10;
+		}
+
+		// Declare needed variables
+		int rotationsCount = 0;
+		boolean numberEqual = false;
+
+		// Check the numbers for equals
+		while ((rotationsCount <= firstNumberLength - 1) && !numberEqual) {
+
+			// Shift digit in first number 
```

Comment:
Can you think of a solution where you don't have to really change the digits places ?